components/camel-google/camel-google-pubsub/src/main/java/org/apache/camel/component/google/pubsub/GooglePubsubComponent.java

```diff
@@ -87,6 +91,11 @@
               description = "How many milliseconds should a producer be allowed to terminate.")
     private int publisherTerminationTimeout = 60000;
 
+    @Metadata(
+              label = "consumer",
+              description = "Additional retryable error codes for synchronous pull. By default the PubSub client library retries ABORTED, UNAVAILABLE, UNKNOWN")
+    private StatusCode.Code[] synchronousPullRetryableCodes = new StatusCode.Code[0];
```

Comment:
I wouldn't use internals classes for Camel option. It makes more sense to set them as Strings and then convert when it's time to use them. If we start going down this path, it will be a mess in all components. So please use a different approach. Thanks.

Reply:
What would be the appropriate Camel exception to throw in case the provided strings cannot be converted to the Enum?
Since the default error codes and the custom codes are both known at start-up, does it make sense to generate the list of retryable codes at component initialization time? Possibly in the init() method?

Reply:
It makes sense to me, but I would wait for @davsclaus feedback on this, because we are focusing a lot of performance/memory footprint and allocation, so maybe we could find a better way

Reply:
We use a String value in other components where we separate the values by comma - this is the most friendly for tooling , spring-boot auto configuration, etc.

Then you can say 

camel.component.google-pubsub.synchronousPullRetryableCodes = DEADLINE_EXCEEDED,RESOURCE_EXHAUSTED

Reply:
Thanks for the feedback, I have updated the PR accordingly. There was another question about where the logic that parses the option should be. In getSubscriberStub() as it is now, or earlier in the component initialization process?